core/src/trezor/ui/layouts/mercury/__init__.py

```diff
@@ -1066,36 +1074,36 @@ async def confirm_ethereum_staking_tx(
         br_name: str = "confirm_ethereum_staking_tx",
         br_code: ButtonRequestType = ButtonRequestType.SignTx,
     ) -> None:
-
-        # intro
-        await confirm_value(
-            title,
-            intro_question,
-            "",
-            br_name,
-            br_code,
-            verb=verb,
-            value_text_mono=False,
-            info_items=(("", address),),
-            info_title=address_title,
-            chunkify_info=chunkify,
-        )
-
-        # confirmation
         if verb == TR.ethereum__staking_claim:
-            items = ((TR.send__maximum_fee, maximum_fee),)
+            summary_items = ((TR.send__maximum_fee, maximum_fee),)
         else:
-            items = (
+            summary_items = (
                 (TR.words__amount, total_amount),
                 (TR.send__maximum_fee, maximum_fee),
             )
-        await _confirm_summary(
-            items=items,
-            title=title,
-            info_title=TR.confirm_total__title_fee,
-            info_items=info_items,
-            br_name=br_name,
-            br_code=br_code,
+        await raise_if_not_confirmed(
+            RustLayout(
+                trezorui2.flow_confirm_output(
+                    title=verb,
+                    subtitle=None,
+                    message=intro_question,
+                    amount=None,
+                    chunkify=False,
+                    text_mono=False,
+                    account=None,
+                    account_path=None,
+                    br_code=br_code,
+                    br_name=br_name,
+                    address=address,
+                    address_title=address_title,
+                    summary_items=summary_items,
+                    fee_items=info_items,
+                    summary_title=verb,
+                    summary_br_name="confirm_total",
+                    summary_br_code=ButtonRequestType.SignTx,
+                    cancel_text=TR.buttons__cancel,  # cancel staking
```

Comment:
What is the comment about?

Reply:
The title of the Cancel screen, according to Figma, should be "Cancel staking" not "Cancel" like the button, so we need to pass 2 separate parameters here, but I got stuck in trying to make the Rust compiler happy with the types, so I went for the compromise of just calling it "Cancel" all over.